src/components/TopicMenu/TopicMenu.js

```diff
@@ -188,6 +188,9 @@ class TopicMenu extends PureComponent {
 
     const collapsed = isCollapsed || activeTopic.key !== topic.key;
 
+    const menuVisibleLayers = (topic.layers || []).filter(l => {
+      return !l.get('hideInLegend') && !l.isMapboxStyleLayer;
```

Comment:
```suggestion
      return !l.get('hideInLegend');
```

Reply:
You don't need to filter just do a find , It will stop the loop when he find the first hideInLegend=false.

Reply:
Good idea use find instead of filter, I implemented